### PR TITLE
Add Debug to Base58PublicKey and add ValidAccountId

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Fixes: #165

### CODE REVIEW

File: near-sdk/src/json_types/account.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+use crate::AccountId;
+
+/// Helper class to validate account ID during serialization and deserializiation
+#[derive(
+    Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize, Serialize,
+)]
+pub struct ValidAccountId(pub AccountId);
+
+impl ValidAccountId {
+    pub fn is_valid(&self) -> bool {
```
Comment (bowenwang1996) [inline near-sdk/src/json_types/account.rs:15]: this doesn't make sense. So something of type `ValidAccountId` can still be invalid?

Comment (evgenykuzyakov) [reply near-sdk/src/json_types/account.rs:15]: I guess I can make it private. In theory I can make inner field private, which will hide the constructor, so you can only construct it from the `TryFrom`.

Comment (bowenwang1996) [reply near-sdk/src/json_types/account.rs:15]: >  In theory I can make inner field private, which will hide the constructor, so you can only construct it from the TryFrom.

Sounds good.

**Comment by nearmax** [comment]
FYI In the past @frol raised an important suggestion that we should make our structures self-validating. E.g. we should not be able to construct an invalid instance of `AccountId` in nearcore.

File: near-sdk/src/json_types/account.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+impl ValidAccountId {
+    fn is_valid(&self) -> bool {
+        is_valid_account_id(&self.0.as_bytes())
+    }
+
+    pub fn account_id(&self) -> &AccountId {
+        &self.0
+    }
+
+    pub fn into_account_id(self) -> AccountId {
```
Comment (nearmax) [inline near-sdk/src/json_types/account.rs:23]: Seems like `impl From<ValidAccountId> for AccountId` already implements this functionality.

Comment (evgenykuzyakov) [reply near-sdk/src/json_types/account.rs:23]: Ok. Should we rename `account_id()` into `.as_ref()`?

**Comment by nearmax** [comment]
Is this PR ready to submit @evgenykuzyakov

**Comment by evgenykuzyakov** [comment]
Yes, will submit after #200  Want to make sure

**Pull request merged**